t/35-script_clone_job_suse.t

```diff
@@ -24,6 +24,12 @@ package Test::FakeLWPUserAgent {
     sub get ($self, $url) { Test::FakeLWPUserAgentMirrorResult->new(is_success => $self->is_validrepo) }
 }    # uncoverable statement
 
+sub create_mock () {
+    my $fake_ua = Test::FakeLWPUserAgent->new;
+    my %url_handler = (remote_url => Mojo::URL->new('http://foo'), ua => $fake_ua);
+    return ($fake_ua, %url_handler);
```

Comment:
I don't think you can return a hash map. This code just expands the hash map again. So this is equivalent to:

```suggestion
    return ($fake_ua, remote_url => Mojo::URL->new('http://foo'), ua => $fake_ua);
```

Probably the following would be better:

```suggestion
    my %url_handler = (remote_url => Mojo::URL->new('http://foo'), ua => $fake_ua);
    return ($fake_ua, \%url_handler);
```

Of course you'll need to change the call-sites to expect a hash ref.

Reply:
> I don't think you can return a hash map.
You can, kind of.
It's correct that it will just be expanded to a list, but in the caller we have:
```
my ($fake_ua, %url_handler) = create_mock;
```
so the returned list will be assigned to a hash again.
But I would also prefer a hash ref in such cases.

Reply:
Yes, but really only "kind of". Try to pass 2 hash maps this way and the card house falls apart :-)